Add CORS origin allow config

k311093 opened the pull request:
Add RunningMode field at conf.toml and add CORS handling when mode is in qa mode.

ser-io commented on conf.toml:
```diff
@@ -1,4 +1,5 @@
 WebStaticFilesPath = "web"
+RunningMode = "prod"
```
we call this concept `environment`. however, using a general `environment` config parameter will result in code like next:

```
if (environment == "qa) {
 // new configs will be set here, from now on
}  
else {
 // or here.
}
``` 

which I think is not good pattern as it makes irrelevant all the existing and new config parameters as you can set them in the `if ... else ...` clause mentioned above. We should rather use an specific config parameters and set the relevant value in each deployment configuration, for example,  in this case we could have something like "EnableCorsException", which could be set true in the config for QA deployment only.

k311093 replied:
Changed to use specific config CORSAllowedOrigins instead of depending on environment name.

k311093 commented:
Changed the config to WebRTC.CorsAllowedOrigins that represents the regex allow list for CORS origins.

Also changed logic for the config WebRTC.CorsAllowedOrigins.

ser-io commented on conf.toml:
```diff
@@ -43,3 +43,7 @@ HostOrchestratorPort = 1081
+
+# Regex patterns of allowed origins
+# e.g. "https://localhost:[0-9]+"
+CorsAllowedOrigins = []
```
Let's use `CORSAllowedOrigins` instead.  https://google.github.io/styleguide/go/decisions.html#initialisms

k311093 replied:
Changed to CORSAllowedOrigins.

ser-io commented on conf.toml:
```diff
@@ -43,3 +43,7 @@ HostOrchestratorPort = 1081
+
+# Regex patterns of allowed origins
+# e.g. "https://localhost:[0-9]+"
+CorsAllowedOrigins = []
```
Let's move the config parameter to the general section to the top below `WebStaticFilesPath` as this one is not a `WebRTC` specific configuration paremeter. In the future, we might add a new config group to list general web related config concepts.

k311093 replied:
Moved CORSAllowedOrigins config from WebRTC section to the top level.

ser-io commented on conf.toml:
```diff
@@ -43,3 +43,7 @@ HostOrchestratorPort = 1081
 [WebRTC]
 STUNServers = ["stun:stun.l.google.com:19302"]
+
+# Regex patterns of allowed origins
```
Having regexes here is too permissive and could become a security issue. 

As we are only tackling CORS from **localhost** we could have an specific **`CORSAllowedLocalhost`** boolean config parameter that allows localhost only. On the other hand, if we need to allow external origins in the future we should list them fully rather than through regex matching.

k311093 replied:
Changed to exact match.

ser-io commented on pkg/app/app.go:
```diff
@@ -68,8 +69,24 @@ func NewApp(
+
+func (c *App) AddCorsHeaderIfNeeded(w http.ResponseWriter, r *http.Request) {
+	origin := r.Header.Get("Origin")
+
```
nit: remove the emtpy lines: 79, 83, 124, 127.

k311093 replied:
Removed Removed unnecessary lines.

ser-io reviewed (approved):
LGTM
